### PR TITLE
feat: annotate models

### LINKED ISSUE #69: Add annotations to models
The enum casting on models is awesome, but it's a bit hard to remember that cast attributes are instances of the enum, rather than a string/int/etc.

Since there's already a command to annotate the enum, maybe we can add another to annotate the model like so?

```php
/**
 * App\Models\User
 *
 * @property \App\Enums\UserTypeEnum|null $user_type
 * @property \App\Enums\RoleEnum|null $role
 */
```

Will PR w/ tests if you are keen :)
Thanks!

### PR DESCRIPTION
**Pull request by atymic**
- Move common annotation logic to abstract
- Switch to more robust docblock editing method
- Update tests to cover new command
- Update Documentation

cc @BenSampo 

When working on this PR I found a large issue with the current method of adding docblocks to classes.

The existing method basically adds a new complete doc comment to the class, irrespective of if one exists or not. 

This means, for example when running the command to annotate all enums in a folder that any enums with existing docblocks will end up with double docblocks.

I've gone ahead and refactored the annotation logic to first parse the existing docblock, add the new tags (replacing any that previously exist for that method/property) and write it back to the class.

This means that any existing docblock descriptions or other annotations will be left in tact.

Let me know your thoughts :)

Closes #69

### CODE REVIEW

File: src/Commands/AbstractAnnotationCommand.php [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            $classDeclaration = "final {$classDeclaration}";
+        } elseif ($reflectionClass->isAbstract()) {
+            $classDeclaration = "abstract {$classDeclaration}";
+        }
+
+        $newDocblock = (new DocBlock\Serializer())->getDocComment($docBlock);
+
+
+        // Remove existing docblock
+        preg_replace(
```
Comment (atymic) [inline src/Commands/AbstractAnnotationCommand.php]: @BenSampo 

I'm not sure if you can suggest a better way of removing the existing docblock.

File: tests/fixtures/annotated_class_two [hunk 2]
```diff
@@ -5,8 +5,8 @@ namespace BenSampo\Enum\Tests\Enums\Annotate;
 use BenSampo\Enum\Enum;
 
 /**
- * @method static static TEST
- * @method static static XYZ
+ * @method static static TEST()
```
Comment (atymic) [inline tests/fixtures/annotated_class_two]: Seems it's correct to have brackets here 
https://docs.phpdoc.org/references/phpdoc/tags/method.html

File: tests/ArtisanCommandsTest.php [hunk 3]
```diff
@@ -45,4 +61,30 @@ public function test_annotate_folder_enums()
...
+
+        $fileSystem->copyDirectory(__DIR__ . '/Models/AnnotateOriginals', __DIR__ . '/Models/Annotate');
+
+        $this->artisan('enum:annotate-model', ['--folder' => __DIR__ . '/Models/Annotate'])->assertExitCode(0);
+
+        $newClass = $fileSystem->get(__DIR__ . '/Models/Annotate/Example.php');
+        $this->assertStringEqualsFile(__DIR__ . '/fixtures/annotated_model_class_one', $newClass);
+    }
+
+    public function test_annotate_model_with_existing_docblock_is_not_changed()
```
Comment (atymic) [inline tests/ArtisanCommandsTest.php]: This test should cover the case where a docblock exists :)

**Comment by atymic** [comment]
Sorry to bug @BenSampo, any thoughts on this?
Would be good to get the command fixed so it doesn't add multiple docblocks 😄

**Comment by BenSampo** [comment]
Merged and tagged!

Thanks for your awesome work on this 👍 🥇 🎉

**Comment by atymic** [comment]
No worries 😄 	
Thanks

**Pull request merged**